File: backend/app/schemas/resume.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ExperienceItem(BaseModel):
    """Represents a work experience entry extracted from a resume."""

    company: str | None = None
    title: str | None = None
    location: str | None = None
    start_date: str | None = None
    end_date: str | None = None
    description: str | None = None
    achievements: list[str] = Field(default_factory=list)
    technologies: list[str] = Field(default_factory=list)


class EducationItem(BaseModel):
    """Represents an education entry extracted from a resume."""

    institution: str | None = None
    degree: str | None = None
    field_of_study: str | None = None
    start_date: str | None = None
    end_date: str | None = None
    description: str | None = None


class ResumeDraft(BaseModel):
    """
    Structured unverified draft extracted from raw resume text by AI.

    NOTE: This is an UNVERIFIED DRAFT. It must be reviewed and confirmed by
    the candidate before being merged into the verified CandidateProfile.
    """

    headline: str | None = None
    summary: str | None = None
    skills: list[str] = Field(default_factory=list)
    experience: list[ExperienceItem] = Field(default_factory=list)
    education: list[EducationItem] = Field(default_factory=list)
    target_titles: list[str] = Field(default_factory=list)


class ResumeResponse(BaseModel):
    """Response schema for returning uploaded resume details."""

    id: UUID
    candidate_profile_id: UUID
    file_name: str
    status: str = "extracted"
    raw_text_length: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ResumeExtractResponse(BaseModel):
    """Response schema for returning the extracted unverified resume draft."""

    id: UUID
    candidate_profile_id: UUID
    file_name: str
    status: str = "draft_generated"
    draft: ResumeDraft

File: frontend/src/types/resume.ts
export interface ExperienceItem {
  company?: string | null;
  title?: string | null;
  location?: string | null;
  start_date?: string | null;
  end_date?: string | null;
  description?: string | null;
  achievements?: string[];
  technologies?: string[];
}

export interface EducationItem {
  institution?: string | null;
  degree?: string | null;
  field_of_study?: string | null;
  start_date?: string | null;
  end_date?: string | null;
  description?: string | null;
}

export interface ResumeDraft {
  headline?: string | null;
  summary?: string | null;
  skills: string[];
  experience: ExperienceItem[];
  education: EducationItem[];
  target_titles: string[];
}

export interface ResumeResponse {
  id: string;
  candidate_profile_id: string;
  file_name: string;
  status: string;
  raw_text_length: number;
  created_at: string;
}

export interface ResumeExtractResponse {
  id: string;
  candidate_profile_id: string;
  file_name: string;
  status: string;
  draft: ResumeDraft;
}
